packages/core/refactor/redux/transactions/actions.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2017–2018, LaborX PTY
+ * Licensed under the AGPL Version 3 license.
+ */
+
+import { ethereumProvider } from '@chronobank/login/network/EthereumProvider'
+import BigNumber from 'bignumber.js'
+import Tx from 'ethereumjs-tx'
+
+export const DUCK_TRANSACTIONS = 'transactions'
+export const TRANSACTIONS_NEW = 'transactions/new'
+export const TRANSACTIONS_REMOVE = 'transactions/remove'
+
+export const sendNewtx = (tx, dao) => async (dispatch, getState) => {
+  const web3 = getState().get('web3')
+
+  const params = dao.getEstimateGasParams(tx)
+
+  const [gasPrice, gasLimit] = await Promise.all([web3.eth.getGasPrice(), web3.eth.estimateGas(params)])
```

Comment:
In case of fail getGasPrice or estimateGas Promise.all will fail at all. Is it OK?